src/gar.lua

```diff
@@ -1905,4 +1888,111 @@ function gar.getPaginatedVaultsForGateway(gatewayAddress, cursor, limit, sortBy,
 	)
 end
 
+--- @param gateway Gateway
+--- @param vaultId WalletAddress | MessageId
+--- @param qty mIO
+--- @param currentTimestamp Timestamp
+--- @param endTimestamp Timestamp
+function gar.createGatewayWithdrawVault(gateway, vaultId, qty, currentTimestamp, endTimestamp)
```

Comment:
nit - we have other "private" functions that aren't exposed to protect state being unintentionally manipulated elsewhere (e.g. `decreaseDelegateStakeAtGateway`). given these operate directly on a gateway object, consider following the same pattern

Reply:
[0db2adb](https://github.com/ar-io/ar-io-network-process/pull/269/commits/0db2adbb4e34397353726a43eb7bb96af9bc9738)